group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordHelpers.java

```diff
@@ -467,6 +471,72 @@ public static Record newEmptyGroupMetadataRecord(
         );
     }
 
+    /**
+     * Creates an OffsetCommit record.
+     *
+     * @param groupId           The group id.
+     * @param topic             The topic name.
+     * @param partitionId       The partition id.
+     * @param offsetAndMetadata The offset and metadata.
+     * @param metadataVersion   The metadata version.
+     * @return The record.
+     */
+    public static Record newOffsetCommitRecord(
+        String groupId,
+        String topic,
+        int partitionId,
+        OffsetAndMetadata offsetAndMetadata,
+        MetadataVersion metadataVersion
+    ) {
+        short version = offsetAndMetadata.expireTimestampMs.isPresent() ?
+            (short) 1 : metadataVersion.offsetCommitValueVersion();
```

Comment:
This corresponds to the logic [here](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L1088).

Reply:
Would it make sense to relocate this logic and the linked logic into MetadataVersion?

Reply:
Yeah, I was debating whether the `offsetAndMetadata.expireTimestampMs.isPresent()` part of this should be in MetadataVersion or not. I could pass a boolean for this purpose.